classes/sota_sanity.bbclass

```diff
@@ -10,6 +10,40 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    if re.match("^[a-zA-Z0-9_-]*$", d.getVar("OSTREE_BRANCHNAME")) is None:
+        status.addresult("OSTREE_BRANCHNAME Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if re.match("^[a-zA-Z0-9_-]*$", d.getVar("SOTA_HARDWARE_ID")) is None:
+        status.addresult("SOTA_HARDWARE_ID Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_CLIENT_FEATURES") not in ("hsm", "secondary-network", ""):
+        status.addresult("SOTA_CLIENT_FEATURES should be set to hsm or secondary-network.\n")
+    if d.getVar("SOTA_CLIENT_PROV") not in ("aktualizr-shared-prov", "aktualizr-device-prov", "aktualizr-device-prov-hsm", ""):
+        status.addresult("Valid options for SOTA_CLIENT_PROV are aktualizr-shared-prov, aktualizr-device-prov, and aktualizr-device-prov-hsm.\n")
+    if re.match("^(https?|ftp|file)://.+$", d.getVar("GARAGE_TARGET_URL")) is None:
+        status.addresult("GARAGE_TARGET_URL is set to a bad url.\n")
+    if re.match("^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$", d.getVar("GARAGE_TARGET_EXPIRES")) is None:
+        status.addresult("GARAGE_TARGET_EXPIRES expects a valid UTC instant.\n")
+    if re.match("^\dY\dM\dD$", d.getVar("GARAGE_TARGET_EXPIRE_AFTER")) is None:
```

Comment:
Will this allow an arbitrary number of years/months/days, or only a single digit for each?